martin/src/bin/martin-cp.rs

```diff
@@ -343,20 +337,53 @@ fn iterate_tiles(tiles: Vec<TileRect>) -> impl Iterator<Item = TileCoord> {
 }
 
 fn check_sources(args: &CopyArgs, state: &ServerState) -> Result<String, MartinCpError> {
-    if let Some(source) = &args.source {
-        Ok(source.clone())
+    if let Some(source_id) = &args.source {
+        Ok(source_id.clone())
     } else {
-        let sources = state.tiles.source_names();
-        if let Some(source) = sources.first() {
-            if sources.len() > 1 {
-                return Err(MartinCpError::MultipleSources(sources.join(", ")));
+        let source_ids = state.tiles.source_names();
+        if let Some(source_id) = source_ids.first() {
+            if source_ids.len() > 1 {
+                return Err(MartinCpError::MultipleSources(source_ids.join(", ")));
             }
-            Ok(source.clone())
+            Ok(source_id.clone())
         } else {
             Err(MartinCpError::NoSources)
         }
     }
 }
+
+fn default_bounds(src: &DynTileSource) -> Vec<Bounds> {
+    if src.sources.is_empty() {
+        vec![Bounds::MAX_TILED]
+    } else {
+        let mut source_bounds = src
+            .sources
+            .iter()
+            .filter_map(|source| source.get_tilejson().bounds)
```

Comment:
🤔 Is this correct?

If a source does not have a bound, I think we need to assume the following

```suggestion
            .map(|source| source.get_tilejson().bounds.unwrap_or_else(|| Bounds::MAX_TILED))
```

what do you think?